### PR TITLE
Paginate listed items

### PR DESCRIPTION
**Pull request by jimmyduty**
To prevent large collections of labels and resources from flooding the chat when listed, display only up to the first N items and provide a keyword argument for displaying additional pages.

Example:

Before:

```
 You: Lita, locker label list
Lita: foo is unlocked
Lita: bar is unlocked
Lita: baz is unlocked
Lita: qux is unlocked
```

After, with the new `config.per_page` set to `3`:

```
 You: Lita, locker label list
Lita: foo is unlocked
      bar is unlocked
      baz is unlocked
      Page 1 of 2 shown. Use --page to specify additional pages.
 You: Lita, locker label list --page 2
Lita: 4 is unlocked
      Page 2 of 2 shown. Use --page to specify additional pages.
```

### CODE REVIEW

File: spec/lita/handlers/locker_labels_spec.rb [hunk 1]
```diff
@@ -41,12 +41,62 @@
...
+      before do
+        robot.config.handlers.locker.per_page = 3
+      end
+
+      it 'includes details about what page was shown' do
+        send_command('locker label create 1')
+        send_command('locker label create 2')
+        send_command('locker label create 3')
+        send_command('locker label create 4')
+        send_command('locker label list')
```
Comment (margueritepd) [inline spec/lita/handlers/locker_labels_spec.rb]: Might be worth putting in the `before` block to reduce duplication?

File: lita-locker.gemspec [hunk 2]
```diff
@@ -19,6 +19,7 @@ Gem::Specification.new do |spec|
   spec.add_runtime_dependency 'lita', '>= 4.2'
   spec.add_runtime_dependency 'redis-objects'
   spec.add_runtime_dependency 'time-lord'
+  spec.add_runtime_dependency 'lita-keyword-arguments'
```
Comment (margueritepd) [inline lita-locker.gemspec]: TIL lita keyword arguments :mindblown:

File: lib/lita/handlers/locker_resources.rb [hunk 3]
```diff
@@ -46,22 +47,49 @@ class LockerResources < Handler
...
+
+          state = resource.state.value
+
+          case state
+          when 'unlocked'
+            unlocked(t('resource.desc', name: key, state: state))
+          when 'locked'
+            locked(t('resource.desc', name: key, state: state))
+          else
+            # This case shouldn't happen, but it will if someone a label
```
Comment (margueritepd) [inline lib/lita/handlers/locker_resources.rb]:
```diff
- someone 
```

File: lib/lita/handlers/locker_resources.rb [hunk 4]
```diff
@@ -46,22 +47,49 @@ class LockerResources < Handler
...
+
+        begin
+          page = Integer(response.extensions[:kwargs][:page].to_s, 10)
+        rescue ArgumentError
+          response.reply t("list.invalid_page_type")
+
+          return
+        end
+
+        pages = (count / config.per_page).ceil + 1
```
Comment (margueritepd) [inline lib/lita/handlers/locker_resources.rb]: This line looks fishy to me. I would feel more confident if I saw tests for edge-cases, like where count was evenly divisible by `per_page`, or where count = 0, count = `per_page`... that kinda thing

File: lib/lita/handlers/locker_labels.rb [hunk 5]
```diff
@@ -12,9 +12,10 @@ class LockerLabels < Handler
       include ::Locker::Resource
 
       route(
-        /^locker\slabel\slist#{COMMENT_REGEX}$/,
+        /^locker\slabel\slist#{COMMENT_REGEX}/,
```
Comment (margueritepd) [inline lib/lita/handlers/locker_labels.rb]: Should we even have `COMMENT_REGEX` here? I'm not sure in what case you would add a comment to the line, and it makes me less certain of how adding a comment would interact with the `--page` keyword 
argument.

Speaking of which, could you describe the `--page` functionality in the `help` text?

File: lib/lita/handlers/locker_labels.rb [hunk 6]
```diff
@@ -54,22 +55,49 @@ class LockerLabels < Handler
...
+          response.reply t("list.invalid_page_type")
 
-            slice.each do |n|
-              l = Label.new(n)
-              response.reply(unlocked(t('label.desc', name: n, state: l.state.value)))
-            end
+          return
+        end
+
+        pages = (count / config.per_page).ceil + 1
```
Comment (margueritepd) [inline lib/lita/handlers/locker_labels.rb]: This code looks pretty similar to the resource labels listing code. Any easy way to remove duplication? (I'm a bit on the fence about it, I generally like waiting until there are 3 users of a piece of code before abstracting, so take my comment with a grain of salt)

File: lib/lita/handlers/locker_labels.rb (hunk 5 above)
Comment (esigler) [reply lib/lita/handlers/locker_labels.rb]: I'm 👍on removing the comment regex from the list command, it was only recently merged to master and hasn't been in a release yet: https://github.com/esigler/lita-locker/pull/52)

**Comment by esigler** [comment]
@jimmyduty - when ready, don't worry too much about the Travis CI tests failing if it doesn't have anything directly to do with your work, I'll fix them up separately and cut a release with your PR.

**Comment by jimmyduty** [comment]
Revised based on feedback. Thanks for the review!

**Comment by jimmyduty** [comment]
Still one RuboCop failure but it's not related to my changes.

**Comment by esigler** [comment]
@jimmyduty - released as 1.1.0
